lib/index.js

```diff
@@ -227,6 +227,23 @@ I18N.prototype.getLocaleSlug = function() {
 	return this.state.localeSlug;
 };
 
+
+/**
+ * Adds new translations to the locale data, overwriting any existing translations with a matching key
+ **/
+I18N.prototype.addTranslations = function( localeData ) {
+	for ( var prop in localeData ) {
```

Comment:
Or use `lodash.merge` and `lodash.omit`:
`merge( this.state.jed.options.locale_data.messages, omit( localeData, '' ) )`